core/server/api/v2/utils/validators/input/posts.js

```diff
@@ -1,7 +1,25 @@
-const _ = require('lodash');
 const Promise = require('bluebird');
 const common = require('../../../../../lib/common');
 const utils = require('../../index');
+const jsonSchema = require('../utils/json-schema');
+
+const handleErrors = (errors) => {
+    // TODO: consider better parsing of `errors` messages
+    // also might be a good idea to move this handling into more central
+    // place once validations are introduced for more endpoints
+    if (['type', 'required'].includes(errors[0].keyword)) {
```

Comment:
It's rather hard to distinguish here errors that should be treated as `common.errors.BadRequestError` from ones that are `common.errors.ValidationError`. Maybe would make sense to handle all of them as `common.errors.ValidationError`? What are your thoughts?

Reply:
Yeah 400 should be actually only used for wrong syntax - e.g. invalid JSON